pkg/operator/resource/resourceapply/apps_test.go

```diff
@@ -134,6 +134,77 @@ func TestApplyDeployment(t *testing.T) {
 	}
 }
 
+func TestDeleteDeployment(t *testing.T) {
+	tests := []struct {
+		name              string
+		desiredDeployment *appsv1.Deployment
+		expectError       bool
+	}{
+		{
+			name:              "when deployment exists",
+			desiredDeployment: workload(),
+			expectError:       false,
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			eventRecorder := events.NewInMemoryRecorder("")
+			fakeKubeClient := fake.NewSimpleClientset()
+			if tt.desiredDeployment != nil {
+				fakeKubeClient = fake.NewSimpleClientset(tt.desiredDeployment)
+			}
+			_, _, err := resourceapply.ApplyDeployment(context.TODO(), fakeKubeClient.AppsV1(), eventRecorder, tt.desiredDeployment, 0)
+			if err != nil {
+				t.Fatalf("unexpected error: %v", err)
+			}
+
+			_, deletedFlag, err := resourceapply.DeleteDeployment(context.TODO(), fakeKubeClient.AppsV1(), eventRecorder, tt.desiredDeployment)
+			if err != nil {
```

Comment:
Didn't you want to use `expectError` here?

Reply:
fixed